app/src/main/java/com/duckduckgo/app/downloads/DownloadsViewModel.kt

```diff
@@ -67,23 +75,28 @@ class DownloadsViewModel @Inject constructor(
         data class CancelDownload(val item: DownloadItem) : Command()
     }
 
-    private val viewState = MutableStateFlow(ViewState())
     private val command = Channel<Command>(1, DROP_OLDEST)
 
-    fun downloads() {
-        viewModelScope.launch(dispatcher.io()) {
-            downloadsRepository.getDownloadsAsFlow().collect {
-                val itemsList = it.mapToDownloadViewItems()
-                viewState.emit(
-                    currentViewState().copy(downloadItems = itemsList, filteredItems = itemsList),
-                )
-            }
-        }
+    private val downloadItems: Flow<List<DownloadViewItem>> = downloadsRepository.getDownloadsAsFlow().map {
+        it.mapToDownloadViewItems()
     }
 
-    fun viewState(): StateFlow<ViewState> {
-        return viewState
-    }
+    private val notifyMeVisibilityChanged = MutableStateFlow(true)
```

Comment:
Should this be false to start with? I think you can avoid the flickering that way. At least, if there's flickering, should be because the Component appears and not because it disappears.